src/Services/Azure/Cosmos/CosmosService.cs

```diff
@@ -14,7 +14,7 @@ public class CosmosService(ISubscriptionService subscriptionService, ITenantServ
     : BaseAzureService(tenantService), ICosmosService, IDisposable
 {
     private const string CosmosBaseUri = "https://{0}.documents.azure.com:443/";
-    private CosmosClient? _cosmosClient;
+    private Dictionary<string, CosmosClient> _cosmosClientCache = [];
```

Comment:
Please use the existing CacheService.  See how Search is doing it.  Thanks

Reply:
https://github.com/Azure/azure-mcp/pull/90/files has that code

Reply:
Updated to follow pattern in Search - but I am not convinced this is the right approach.

The CosmosClient is meant to be a long lived object that should be around for the duration of the execution so tying it's lifecycle to the calling service seemed appropriate. 

When using the cache, we do not control what happens with the object when it gets evicted so we don't get an opportunity to properly dispose. 
If we set a very long expiration (essentially never evict) then we have an issue trying to find all the clients in the cache so we can dispose them properly when the base CosmoService gets disposed. (see empty Dispose method now)

Looking for guidance on how to address this!
Thanks

Reply:
Could you provide more details about
"we have an issue trying to find all the clients in the cache so we can dispose them properly when the base CosmosService gets disposed."?

In practical situations, shouldn't there be a limited number of cached clients?

Reply:
Since the user can ask the about multiple Cosmos Accounts, we will have a CosmosClient instance for each account the user is asking about. We will cache every CosmosClient based on the account name ("cosmos_clients_account1", "cosmos_clients_account2" , etc..)

When we cache these clients locally in a dictionary, it's easy to dispose them all since we easily query the dictionary. 
In the absence of a method that can return list of objects based on key prefix on ICacheClient it will be difficult to find all Clients in the Cache in order to dispose them.

Reply:
I think this is a valid request.

Let's go ahead to merge the fix for //build and I opened https://github.com/Azure/azure-mcp/issues/139 to track the request.

Reply:
Let's do this in a generic way so that other clients can benefit.  If the CacheService doesn't meet needs, then how can we update to meet them?